openedx/core/djangoapps/courseware_api/utils.py

```diff
@@ -66,3 +72,42 @@ def serialize_upgrade_info(user, course_overview, enrollment):
         'sku': mode.sku,
         'upgrade_url': verified_upgrade_deadline_link(user, course_overview),
     }
+
+
+def get_learning_assistant_launch_url(user, course_key, enrollment_object, overview):
+    """
+    Return the launch URL for an LTI based learning assistant
+    """
+    config_id = getattr(settings, 'LEARNING_ASSISTANT_CONFIG_ID', None)
+    if not config_id:
+        return None
+
+    try:
+        lti_config = LtiConfiguration.objects.get(config_id=config_id)
+    except LtiConfiguration.DoesNotExist:
+        return None
+
+    if lti_config.version != 'lti_1p3':
+        return None
+
+    user_role = get_user_role(user, course_key)
+    if (
+        (enrollment_object and enrollment_object.mode not in CourseMode.VERIFIED_MODES)
+        and user_role != 'staff'
```

Comment:
Should instructors be able to load the tool as well?

Reply:
Let me check with Spencer!